src/Handlers/UserHandler.php

```diff
@@ -54,12 +63,17 @@ public function getUserListPaginated(
             $userListPagination->pageSize = $numResults;
         }
 
+        $numPages = (int)ceil($numResults / $userListPagination->pageSize);
+        if (!$this->parameterValidator->isParameterValid($numPages, $userListPagination->currentPage)) {
+            throw new NotFoundHttpException();
+        }
```

Comment:
The block `$context = ...` and `$json = $this->serializer...` should be after this validation. Check all the controllers.